### PR TITLE
ref(similarity-embedding): Add would group to bulk analytics

### PR DESCRIPTION
**Pull request by jangjodi**
Add whether the similarity algorithm would've grouped the issues to the bulk agree/disagree analytics

### CODE REVIEW

File: static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx [hunk 1]
```diff
@@ -63,6 +67,9 @@ class SimilarToolbar extends Component<Props, State> {
           parentGroupId,
           groupId: this.props.groupId,
           value,
+          wouldGroup: itemWouldGroup?.length
+            ? itemWouldGroup[0].shouldBeGrouped
+            : undefined,
```
Comment (scttcper) [inline static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx:72]: nit should be the same
```suggestion
          wouldGroup: itemWouldGroup[0]?.shouldBeGrouped,
```

File: static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx [hunk 2]
```diff
@@ -55,6 +56,9 @@ class SimilarToolbar extends Component<Props, State> {
       return;
     }
     for (const parentGroupId of this.state.mergeList) {
+      const itemWouldGroup = this.props.itemsWouldGroup
+        ? this.props.itemsWouldGroup.filter(item => item.id === parentGroupId)
+        : undefined;
```
Comment (scttcper) [inline static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx:61]: same here, can use `?` to optional chain. Also if we're only using the first item, prefer using find over filter
```suggestion
      const itemWouldGroup = this.props.itemsWouldGroup?.find(item => item.id === parentGroupId)
```

File: static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ import {trackAnalytics} from 'sentry/utils/analytics';
 type Props = {
   onMerge: () => void;
   groupId?: string;
+  itemsWouldGroup?: Array<Record<string, string | undefined>> | null;
```
Comment (scttcper) [inline static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx:18]: be more specific on the type
```suggestion
  itemsWouldGroup: Array<{id: string, shouldBeGrouped: boolean}> | null;
```

Comment (jangjodi) [reply static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/toolbar.tsx]: Thanks for the feedback!

**Review by scttcper — APPROVED** [review]

**Pull request merged**